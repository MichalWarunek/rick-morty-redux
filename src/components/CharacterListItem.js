import React from 'react';
import { Col } from 'reactstrap';
import { Link } from 'react-router-dom';

const CharacterListItem = ({id, image, name, status, species, gender }) => (
 

  <Col md="2" sm="4" xs="6">
    <Link to={`/edit/${id}`}>
      <img className="item-image" src={image}/>
    </Link>
    <div className="item">
    <h3>{name}</h3>
      <div className="show-for-desktop">
        <div className="item-properties"> 
          <div>
            <h4>status</h4> 
          </div>
          <div className="item-property">
            <h4>{status.toLowerCase()}</h4> 
            </div>
          </div>
          <div className="item-properties"> 
            <div>
              <h4>species</h4>
            </div>
          <div className="item-property">
            <h4>{species.toLowerCase()}</h4>
            </div>
          </div>
          <div className="item-properties"> 
          <div>
            <h4>gender</h4> 
          </div>
          <div className="item-property">
            <h4>{gender.toLowerCase()}</h4> 
            </div>
          </div>
        </div>
    </div>
  </Col>
);

export default CharacterListItem;