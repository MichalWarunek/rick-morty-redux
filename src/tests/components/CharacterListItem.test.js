import React from 'react';
import { shallow } from 'enzyme';
import CharacterListItem from '../../components/CharacterListItem';

test('should correctly render character list item', () => {
  const wrapper = shallow(<CharacterListItem />);
  expect(wrapper).toMatchSnapshot();
});