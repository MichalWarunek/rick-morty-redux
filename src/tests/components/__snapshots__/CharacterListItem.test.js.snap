// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render character list item 1`] = `
<div>
  This is CharacterListItem
</div>
`;
